### PR TITLE
Feat: CLI buckets init&deploy

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

As implemented by init&deploy collection, this PR adds the same logic for buckets. It' common practice to have a predefined structure of buckets, just like it's common for databases. Since appwrite.json is supposed to be a source of truth for initial setup of Appwrite project, it should include structure of buckets.

This PR gets us one small step closer to "Deploy to Appwrite" 1-click setup.

## Test Plan

- [x] Manual QA

https://user-images.githubusercontent.com/19310830/193419300-2ba13066-6e37-4b86-96b7-676c10ccd12d.mp4

## Related PRs and Issues

x

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: templates/cli/lib/commands/deploy.js.twig [hunk 1]
```diff
@@ -593,6 +596,93 @@ const deployCollection = async ({ all, yes } = {}) => {
...
+    const configBuckets = localConfig.getBuckets();
+
+    if(all) {
+        if (configBuckets.length === 0) {
+            throw new Error("No buckets found in the current directory. Run `appwrite init bucket` to fetch all your buckets.");
+        }
+        bucketIds.push(...configBuckets.map((b) => b.$id));
+    }
+
+    if(bucketIds.length <= 0) {
```
Comment (lohanidamodar) [inline templates/cli/lib/commands/deploy.js.twig:612]: is it ever possible to be `< 0` ?

**Review by lohanidamodar — COMMENTED** [review]
Looks good, just a small query

File: templates/cli/lib/commands/deploy.js.twig (hunk 1 above)
Comment (Meldiron) [reply templates/cli/lib/commands/deploy.js.twig:612]: Nah. It's just a way I do empty array check. Should I do `.length === 0` instead?

Comment (lohanidamodar) [reply templates/cli/lib/commands/deploy.js.twig:612]: I think this makes more sense

**Review by lohanidamodar — APPROVED** [review]

**Comment by lohanidamodar** [comment]
@Meldiron can you work on getthing the relevant docs updated for this change?

**Comment by Meldiron** [comment]
@lohanidamodar resolved

**Pull request merged**